unit-tests/live/frames/lrs_frame_queue_manager.py

```diff
@@ -0,0 +1,235 @@
+
+import pyrealsense2 as rs
+
+
+class ModifiedLogger(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        return "{serial}: {msg}".format(serial=self.extra['serial'], msg=msg), kwargs
+
+
+class LRSFrameQueueManager:
+    """Manage a queue structure for drop-safe and high-performance frame handling using pyrealsense.
+
+    The queue structure consist of 2 queues: lrs frame_queue and python threading queue.
+    The lrs frame_queue is filled with frames arriving from lrs while the threading queue get frames from it
+    and send it for post-process.
+
+    Each frame which is inserted to the lrs.frame_queue is marked with keep=True
+    to prevent frame drops due to slow frame processing.
+    Frames from the lrs frame_queue are sent for post-process using producer-consumer threads;
+    the producer thread get frames from the lrs frame_queue and put it in the threading queue, while
+    the consumer thread get frames from the threading queue and sent it for post-process.
+
+    Attributes:
+        lrs_queue: pyrealsesne.frame_queue through it frames arrive from lrs
+
+    Args:
+        callback (function): Post-process callback to be called on each frame in the post process queue.
+        max_queue_size (int): Max size of a queue in the frame queues structure.
+        statistics (bool): Whether to collect performance statistics or not.
+        If collected, will be saved in a CSV file under '~/queue_<ts>.csv
+        serial_number (str): The serial number of the camera of which the queues belongs to.
+    """
+
+    class Event(object):
+        TERMINATE_CONSUMER = 'terminate_consumer'
+        START_COLLECTING = 'start_collecting'
+        STOP_COLLECTING = 'stop_collecting'
+
+    def __init__(self, callback=None, max_queue_size=100000, statistics=False, serial_number=None):
+        self._id = serial_number
+        self._max_queue_size = max_queue_size
+
+        self._logger = ModifiedLogger(logging.getLogger('test'), {'serial': self._id}, )
+        self._process = callback
+
+        self.lrs_queue = rs.frame_queue(capacity=self._max_queue_size, keep_frames=True)
+
+        self._post_process_queue = Queue(maxsize=self._max_queue_size)
+        self._producer_thread = None
+        self._consumer_thread = None
+        self._block_queue_event = threading.Event()
+        self.block()  # ignoring frames until self.start is called
+        self._terminate_producer_event = threading.Event()
+
+        self.statistics = statistics
+
+        self._producing_times = collections.deque()
+        self._producer_queue_sizes = collections.deque()
+        self._consuming_times = collections.deque()
+        self._consumer_queue_sizes = collections.deque()
+        self._memory_samples = collections.deque()
+
+    def start(self):
+        """Initialize the queue structure and starting the producer-consumer threads
+
+        first, unblocking the threading queue and clear all thread's events and then
```

Comment:
In the test above the following code appears.
```python
    lrs_fq.start()
    lrs_fq.unblock()
```
Not clear why unblock is not part of start, I see that inside  stop()  function there is a call for block().
It makes sense to call unblock internally (atomic).

Reply:
fixed